### PR TITLE
Fixes #32205 - Add an option to keep the ansible-runner files on Satellite

### PR DESCRIPTION
**Pull request by adiabramovitch**
related to https://github.com/theforeman/foreman_ansible/pull/455

### CODE REVIEW

**Comment by xprazak2** [comment]
Good start.

I am getting an error when the parameter is not added to host. It is possible to specify a default return value when using `fetch`:

no-key-found

Also I think it might be better to invert the flag - clean up the files unless the host parameter requests to keep them.

File: lib/smart_proxy_ansible/runner/ansible_runner.rb [hunk 1]
```diff
@@ -23,6 +23,7 @@ def initialize(input, suspended_action:)
         @tags_flag = action_input[:tags_flag]
         @passphrase = action_input['secrets']['key_passphrase']
         @execution_timeout_interval = action_input[:execution_timeout_interval]
+        @cleanup_working_dirs = input.fetch(:cleanup_working_dirs)
```
Comment (ezr-ondrej) [inline lib/smart_proxy_ansible/runner/ansible_runner.rb:26]: Ondra's right, it's better to have a fallback to default value, just to be safe
```suggestion
        @cleanup_working_dirs = input.fetch(:cleanup_working_dirs, true)
```

File: lib/smart_proxy_ansible/runner/ansible_runner.rb [hunk 2]
```diff
@@ -70,7 +71,7 @@ def kill
 
       def close
         super
-        FileUtils.remove_entry(@root) if @tmp_working_dir && Dir.exist?(@root)
+        FileUtils.remove_entry(@root) if @tmp_working_dir && Dir.exist?(@root) if @cleanup_working_dirs
```
Comment (ezr-ondrej) [inline lib/smart_proxy_ansible/runner/ansible_runner.rb:74]: Double `if` feels weird
```suggestion
        FileUtils.remove_entry(@root) if @tmp_working_dir && Dir.exist?(@root) && @cleanup_working_dirs
```

**Comment by ezr-ondrej** [comment]
Thanks @adiabramovitch !

**Pull request merged**